src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -486,7 +486,17 @@ void WindowEmperor::HandleCommandlineArgs(int nCmdShow)
 
             if (msg.message == WM_KEYDOWN)
             {
-                IslandWindow::HideCursor();
+                const bool modifierKey = (msg.wParam == VK_SHIFT ||
+                                          msg.wParam == VK_CONTROL ||
+                                          msg.wParam == VK_MENU ||
+                                          msg.wParam == VK_LWIN ||
+                                          msg.wParam == VK_RWIN);
+
+                // Hide the cursor only when the key pressed is not a modifier key.
+                if (!modifierKey)

```

Comment:
Thank you for working on this. Elsewhere we have this function:

https://github.com/microsoft/terminal/blob/b8f35a31c26febbe153642488c88549f93e5980c/src/cascadia/TerminalCore/Terminal.hpp#L66-L80

Perhaps we should copy it verbatim here and use it?